### ISSUE TITLE
@angular-eslint v18 requires typescript-eslint v8 but not v7

### ISSUE THREAD

**Issue by pkurcx** [angular-eslint/angular-eslint#1853]
It would be nice to have a possibility to use typescript-eslint v7 as described in v18 release 

https://github.com/angular-eslint/angular-eslint/blob/c8dff2faf86e3cbfcf9003888c306f020a3937f0/packages/utils/package.json#L25

**Comment by JamesHenry** [comment]
The new `angular-eslint` package is for ESLint v9, typescript-eslint v8 and Flat Config. If you do not want to use that combination then do not use the new package. typescript-eslint v8 is where full support for ESLint v9 was landed.

You can use typescript-eslint v7 with angular-eslint v18 so I am closing this one

**Comment by pkurcx** [comment]
@JamesHenry 

How can I handle this?

<img width="948" alt="Screenshot 2024-05-29 at 20 37 02" src="https://github.com/angular-eslint/angular-eslint/assets/11500742/4c635a8b-5929-4fed-87d0-90c4d7488042">

<img width="291" alt="Screenshot 2024-05-29 at 20 38 08" src="https://github.com/angular-eslint/angular-eslint/assets/11500742/438ec46f-c926-4baa-a5ad-84ff31ae5ee4">

**Comment by JamesHenry** [comment]
Ah ok the utils package, I apologise I misread the link above initially.

Please can you share more about your setup? Just so I can build a test repo to make sure any changes I make are appropriate.

I imagine just adding the v7 as an `||` will work.

But yeah please can you share your package.json deps and and example of the usage?

**Comment by pkurcx** [comment]
Here you are:

```json
{
  "name": "frontend-libraries",
  "version": "0.0.0",
  "scripts": {
   
  },
  "engines": {
    "node": ">=20.11.0 <21.0.0",
    "npm": ">=10.2.0 <11.0.0"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "18.0.1",
    "@angular/cdk": "18.0.0",
    "@angular/common": "18.0.1",
    "@angular/compiler": "18.0.1",
    "@angular/core": "18.0.1",
    "@angular/forms": "18.0.1",
    "@angular/material": "18.0.0",
    "@angular/platform-browser": "18.0.1",
    "@angular/platform-browser-dynamic": "18.0.1",
    "@angular/router": "18.0.1",
    "@nx/angular": "19.1.0",
    "rxjs": "7.8.1",
    "tslib": "2.6.2",
    "tsutils": "3.21.0",
    "zone.js": "0.14.6"
  },
  "devDependencies": {
    "@angular-devkit/architect": "0.1800.2",
    "@angular-devkit/build-angular": "18.0.2",
    "@angular-devkit/core": "18.0.2",
    "@angular-devkit/schematics": "18.0.2",
    "@angular-eslint/builder": "18.0.0",
    "@angular-eslint/eslint-plugin": "18.0.0",
    "@angular-eslint/eslint-plugin-template": "18.0.0",
    "@angular-eslint/schematics": "18.0.0",
    "@angular-eslint/template-parser": "18.0.0",
    "@angular/cli": "18.0.2",
    "@angular/compiler-cli": "18.0.1",
    "@angular/language-service": "18.0.1",
    "@nx/cypress": "19.1.0",
    "@nx/eslint": "19.1.0",
    "@nx/eslint-plugin": "19.1.0",
    "@nx/jest": "19.1.0",
    "@nx/js": "19.1.0",
    "@nx/node": "19.1.0",
    "@nx/workspace": "19.1.0",
    "@schematics/angular": "18.0.1",
    "@types/eslint": "8.56.3",
    "@types/jest": "29.5.12",
    "@types/node": "20.11.20",
    "@typescript-eslint/eslint-plugin": "7.11.0",
    "@typescript-eslint/parser": "7.11.0",
    "@typescript-eslint/rule-tester": "7.11.0",
    "@typescript-eslint/typescript-estree": "7.11.0",
    "@typescript-eslint/utils": "7.11.0",
    "autoprefixer": "10.4.17",
    "eslint": "8.57.0",
    "eslint-config-prettier": "9.1.0",
    "eslint-plugin-ban": "^1.6.0",
    "eslint-plugin-import": "2.29.1",
    "eslint-plugin-no-only-tests": "^3.1.0",
    "eslint-plugin-prefer-arrow": "^1.2.3",
    "eslint-plugin-rxjs": "5.0.3",
    "eslint-plugin-rxjs-angular": "2.0.1",
    "jest": "29.7.0",
    "jest-environment-jsdom": "29.7.0",
    "jest-image-snapshot": "6.4.0",
    "jest-preset-angular": "14.1.0",
    "ng-packagr": "18.0.0",
    "nx": "19.1.0",
    "patch-package": "8.0.0",
    "postcss": "8.4.35",
    "postcss-import": "16.0.1",
    "postcss-preset-env": "9.4.0",
    "postcss-url": "10.1.3",
    "sass": "1.72.0",
    "ts-jest": "29.1.2",
    "ts-node": "10.9.2",
    "typescript": "5.4.5"
  }
}
```

```typescript
import {ASTUtils, Selectors} from '@angular-eslint/utils';
import {ESLintUtils, TSESTree} from '@typescript-eslint/utils';
import {AST_NODE_TYPES} from '@typescript-eslint/types';
import {defaultModulesToWarn} from '../defaults';

import {ASTUtils, Selectors} from '@angular-eslint/utils';
import {ESLintUtils, TSESTree} from '@typescript-eslint/utils';
import {AST_NODE_TYPES} from '@typescript-eslint/types';
import {defaultModulesToWarn} from '../defaults';

export const rule = ESLintUtils.RuleCreator(name => createUrl(name))({
  name: 'warn-wrong-module',
  meta: {
    messages: {
      warnWrongModule: 'msg',
      replaceWith: 'rpl',
    },
    docs: {
      description: 'Warn about possibly wrong module import',
      recommended: undefined,
    },
    schema: [
      {
        type: 'object',
        properties: {
          shouldAppendToExisting: {
            type: 'boolean',
          },
          modulesToWarn: {
            type: 'array',
            items: {
              type: 'object',
              properties: {
                moduleToWarn: {
                  type: 'string',
                },
                replacement: {
                  type: 'string',
                },
              },
              required: ['moduleToWarn'],
            },
          },
        },
        additionalProperties: false,
      },
    ],
    type: 'problem',
    hasSuggestions: true,
  },
  defaultOptions: [
    {
      shouldAppendToExisting: true,
      modulesToWarn: [] as {moduleToWarn: string; replacement: string}[],
    },
  ],
  create(context, [{shouldAppendToExisting, modulesToWarn}]) {
    const modulesToCheck = shouldAppendToExisting ? [...modulesToWarn, ...defaultModulesToWarn] : modulesToWarn;

    function checkImports(node: TSESTree.Decorator) {
      const importsPropertyValue = ASTUtils.getDecoratorPropertyValue(node, 'imports');
      if (!importsPropertyValue || importsPropertyValue.type !== AST_NODE_TYPES.ArrayExpression) {
        return;
      }
      const imports = (importsPropertyValue.elements as TSESTree.Expression[]).filter(isIdentifier);
      const moduleName = ASTUtils.getClassName(node.parent as TSESTree.ClassDeclaration); // parent type ensured by selector
      // impl
    }

    return {
      [Selectors.MODULE_CLASS_DECORATOR]: checkImports,
      [Selectors.COMPONENT_CLASS_DECORATOR]: checkImports,
    };
  },
});

```

**Comment by pkurcx** [comment]
Yes, I guess `|| v7` should do the work. Please verify all packages, cause not only `utils` lacks `|| v7`

**Comment by JamesHenry** [comment]
Yeah I will but only the utils use-case should be subject to this type mismatch, because in the other scenarios the packages are not consumed in code by the user. Will put up a resolution tomorrow.

Package manager overrides/resolutions should unblock you in the immediate term?

**Comment by JamesHenry** [comment]
This should be resolved with `18.0.1`, sorry again for the confusion
